### PR TITLE
clang-tidy bugprone fixes

### PR DESCRIPTION
**Pull request by neheb**

### CODE REVIEW

**Comment by neheb** [comment]
Brilliant. The last commit and the Windows debug builder found a significant issue in src/preview.cpp . For some laughs: https://youtu.be/yG1OZ69H_-o?t=2357

File: src/pngimage.cpp [hunk 1]
```diff
@@ -241,7 +241,7 @@ namespace Exiv2
             const long imgSize = static_cast<long>(io_->size());
             DataBuf cheaderBuf(8);
 
-            while (!io_->eof() && ::strcmp(chType, "IEND")) {
+            while (!io_->eof() && ::strcmp(chType, "IEND") != 0) {
```
Comment (piponazo) [inline src/pngimage.cpp:244]: :clap:

File: src/http.cpp [hunk 2]
```diff
@@ -44,10 +44,6 @@
 #define SLEEP 1000
 #define SNOOZE 0
 
-#ifdef __MINGW__
-#define fopen_S(f, n, a) f = fopen(n, a)
```
Comment (piponazo) [inline src/http.cpp:48]: I was wondering why this dissapeared and after checking the code I noticed that it is not used at all. Is it something that clang-tidy deleted automatically or was it a manual change?

**Review by piponazo — APPROVED** [review]

File: src/http.cpp (hunk 2 above)
Comment (neheb) [reply src/http.cpp:48]: It was manual. However clang-tidy does warn about this in one of its checks. It recommends doing away with function macros and using constexpr functions (mostly the same behavior).

**Pull request merged**